hal/src/electron/modem/mdm_hal.cpp

```diff
@@ -1732,6 +1760,45 @@ int MDMParser::_cbCNUM(int type, const char* buf, int len, char* num)
     return WAIT;
 }
 
+int MDMParser::_cbUCGED(int type, const char* buf, int len, NetStatus* status)
+{
+    if ((type == TYPE_PLUS || type == TYPE_UNKNOWN) && status) {
+        int rsrp;
+        int rsrq_n;
+        unsigned long rsrq_f;
+        // AT+UCGED?
+        // +RSRP: 314,5110,"-102.60",163,5110,"-097.90",173,5110,"-100.40",
+        // +RSRQ: 314,5110,"-19.60",163,5110,"-16.60",173,5110,"-18.60",
+
+        // RSRP maps from dBm [-140,-44] to [0,97]
+        if (sscanf(buf, "\r\n+RSRP: %*d,%*d,\"%d\"",&rsrp) == 1){
+            if (rsrp < -140) {
+                status->rsrp = 0;
+            }
+            else if (rsrp >= -44) {
+                status->rsrp = 97;
+            }
+            else {
+                status->rsrp = rsrp + 140;
+            }
+        }
+        // RSRQ maps from dBm [-20,-3] to [0,34]
+        if (sscanf(buf, "\r\n+RSRQ: %*d,%*d,\"%d.%lu\"",&rsrq_n, &rsrq_f) == 2){
+            int rsrq_mul_100 = rsrq_n * 100 - rsrq_f;
+            if (rsrq_mul_100 < -19.5 * 100) {
```

Comment:
Nitpicking: this will be optimized by the compiler anyway, but probably better to have a non-floating point constant.